lib/fastly_nsq/message_queue/producer.rb

```diff
@@ -1,32 +1,33 @@
+require 'forwardable'
+
 class InvalidParameterError < StandardError; end
 
 module MessageQueue
   class Producer
-    def initialize(topic:, ssl_context: nil)
-      @topic = topic
-      @ssl_context = SSLContext.new(ssl_context)
-    end
+    extend Forwardable
+    def_delegator :connection, :terminate
+    def_delegator :connection, :write
 
-    def connection
-      @producer ||= producer.new(params)
-    end
-
-    def terminate
-      @producer.terminate
+    def initialize(topic:, ssl_context: nil, &connector)
```

Comment:
Same comment as [here], I'm a very gentle -1 on the constructor taking a block, although I don't have an alternative approach, which is not that helpful. The delegation stuff I'm a big 👍  on.

[here]: https://github.com/fastly/fastly_nsq/pull/32/files#r62891135